processes/ingest/loc.py

```diff
@@ -1,290 +1,114 @@
-                    #Weekly/Custom Ingestion Conditional
-                    if customTimeStamp:
-                        itemTimeStamp = datetime.strptime(metaDict['timestamp'], '%Y-%m-%dT%H:%M:%S.%fZ')
-
-                        if itemTimeStamp < customTimeStamp:
-                            whileBreakFlag = True
-                            break
+        start_datetime = utils.get_start_datetime(
+            process_type=self.process_type,
+            ingest_period=self.ingest_period
+        )
 
-                    if 'resources' in metaDict.keys():
-                        if metaDict['resources']:
-                            resources = metaDict['resources'][0]
-                            if 'pdf' in resources.keys() or 'epub_file' in resources.keys():
-                                logger.debug(f'OPEN ACCESS URL: {openAccessURL}')
-                                logger.debug(f"TITLE: {metaDict['title']}")
+        records = self.loc_service.get_records(
+            start_timestamp=start_datetime,
+            limit=self.limit,
+        )
 
-                                self.processLOCRecord(metaDict)
-                                count += 1
+        for record in records:
+            self.store_pdf_manifest(record=record)
+            self.store_epub(record=record)
 
-                                logger.debug(f'Count for OP Access: {count}')
+            self.record_buffer.add(record=record)
 
-                if whileBreakFlag == True:
-                    logger.debug('No new items added to collection')
-                    break
+        self.record_buffer.flush()
 
-                sp += 1
-                time.sleep(5)
+        logger.info(f'Ingested {self.record_buffer.ingest_count} LOC records')
 
-        except Exception or HTTPError or IndexError or KeyError as e:
-            logger.exception(e)
+    def store_pdf_manifest(self, record: Record):
+        record_id = record.source_id.split('|')[0]
+        pdf_part = next(filter(lambda part: part.file_type == 'application/pdf', record.get_parts()), None)
 
-        return count
+        if pdf_part is not None:
+            manifest_path = f'manifests/{record.source}/{record_id}'
+            manifest_url = get_stored_file_url(storage_name=self.s3_bucket, file_path=manifest_path)
+            manifest_json = self.generate_manifest(record=record, source_uri=pdf_part.url, manifest_uri=manifest_url)
 
-    def importDigitizedRecords(self, count, customTimeStamp):
-        sp = 1
-        try:
+            self.s3_manager.createManifestInS3(manifestPath=manifest_path, manifestJSON=manifest_json, s3_bucket=self.s3_bucket)
 
-            whileBreakFlag = False
+            record.has_part.insert(0, Part(
+                index=pdf_part.index,
+                url=manifest_url,
+                source=record.source,
+                file_type='application/webpub+json',
+                flags=FileFlags(reader=True).to_string()
+            ).to_string())
```

Comment:
In my recent commit to refactor the manifest methods, this store_pdf_manifest method is called from the s3 manager if you merge in changes from the main branch the LOCProcess will just call the store_pdf_manifest method from the s3Manager.

Reply:
good call - ill merge in hopefully not too many merge conflicts!